rollbar-java/src/main/java/com/rollbar/notifier/config/ConfigBuilder.java

```diff
@@ -471,6 +485,42 @@ public ConfigBuilder truncateLargePayloads(boolean truncate) {
     return this;
   }
 
+  /**
+   * <p>
+   * Maximum Telemetry Events sent in a payload (This value can be between 1 and 50, exceed any of
+   * these thresholds and the closest will be taken). Default: 10.
+   * </p>
+   * @param maximumTelemetryData max quantity of telemetry events sent.
+   * @return the builder instance.
+   */
+  public ConfigBuilder maximumTelemetryData(int maximumTelemetryData) {
+    if (maximumTelemetryData < MINIMUM_CAPACITY_FOR_TELEMETRY_EVENTS) {
+      this.maximumTelemetryData = MINIMUM_CAPACITY_FOR_TELEMETRY_EVENTS;
+      return this;
+    }
+
+    if (maximumTelemetryData > MAXIMUM_CAPACITY_FOR_TELEMETRY_EVENTS) {
+      this.maximumTelemetryData = MAXIMUM_CAPACITY_FOR_TELEMETRY_EVENTS;
+      return this;
+    }
+
+    this.maximumTelemetryData = maximumTelemetryData;
+    return this;
+  }
+
+  /**
+   * <p>
+   * Set a {@link TelemetryEventTracker} implementation.
+   * Default: {@link RollbarTelemetryEventTracker} with a {@link TimestampProvider}.
+   * </p>
+   * @param telemetryEventTracker true to enable truncation.
+   * @return the builder instance.
+   */
+  public ConfigBuilder telemetryEventTracker(TelemetryEventTracker telemetryEventTracker) {
```

Comment:
Telemetry event capacity limits are not applied when you pass a pre-configured `TelemetryEventTracker` here.

Reply:
that's because 60 is over the maximum capacity
![Captura de pantalla 2024-09-25 a la(s) 7 19 25 p  m](https://github.com/user-attachments/assets/20b96ea0-779f-4014-9a71-2582f9658779)

Reply:
the Telemetry event capacity is applied, but the maximumTelemetryData in `ConfigBuilder` is unused
![Captura de pantalla 2024-09-25 a la(s) 7 45 07 p  m](https://github.com/user-attachments/assets/b1150912-37a2-462c-9ce6-ef44cd1a5a99)


`ConfigBuilder` now has two new fields **maximumTelemetryData** and **telemetryEventTracker**
**maximumTelemetryData** would be only for the default TelemetryEventTracker
if a user add a TelemetryEventTracker the **maximumTelemetryData** is unused
so the maximum capacity you set is applied

but, I think it shouldn't be the case, because I wanted to limit the size of TelemetryEvents to 50
![Captura de pantalla 2024-09-25 a la(s) 7 47 05 p  m](https://github.com/user-attachments/assets/fc2cec5c-6e07-42d5-8aec-1afc1596c59b)

I can throw an IllegalArgumentException for that in the RollbarTelemetryEventTracker and remove maximumTelemetryData from `ConfigBuilder`, do you think that's ok?

Reply:
Do we have specs/requirements for limiting telemetry events? If so, let's make sure it is applied consistently everywhere. Otherwise, it can be removed. Throwing an exception during a setup step is not great, especially if the alternative is to fallback to a pre-defined constant.